File: Minah7/week03/캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        LinkedList<String> cache = new LinkedList<>();
        for(int i = 0; i < cities.length; i++) {
            String c = cities[i].toLowerCase();
            if(!cache.contains(c)) {
                cache.addFirst(c);
                answer += 5;
            } else {
                cache.remove(c);
                cache.addFirst(c);
                answer ++;
            }
            if(cache.size() > cacheSize) {
                cache.pollLast();
            }
        }
        return answer;
    }
}

File: nijesmik/week07/캐시.js
const CACHE_HIT = 1;
const CACHE_MISS = 5;

function solution(cacheSize, cities) {
    if (cacheSize === 0) {
        return cities.length * CACHE_MISS;
    }

    const cache = Array(cacheSize);
    let time = 0;
    for (let i = 0; i < cities.length; i++) {
        time += search(cache, cities[i].toLowerCase());
    }
    return time;
}

const search = (cache, city) => {
    for (let i = 0; i < cache.length; i++) {
        if (cache[i] === city) {
            cache.splice(i, 1);
            cache.push(city);
            return CACHE_HIT;
        }
    }
    cache.shift();
    cache.push(city);
    return CACHE_MISS;
}
